Create CONTRIBUTING.md (#741)
epinio/epinio: CONTRIBUTING.md

# Contributing to Epinio

> Rancher Desktop accepts contributions via GitHub issues and pull requests.

Reviewer: ```suggestion
Epinio accepts contributions via GitHub issues and pull requests.
```

This document outlines the process to get your pull request accepted.

## Start With An Issue

Prior to creating a pull request it is a good idea to [create an issue].
This is especially true if the change request is something large.
The bug, feature request, or other type of issue can be discussed prior to
creating the pull request. This can reduce rework.

> [create an issue]: https://github.com/rancher-sandbox/rancher-desktop/issues/new

Reviewer: ```suggestion
[Create an issue]: https://github.com/epinio/epinio/issues/new
```

## Sign Your Commits

A sign-off is a line at the end of the explanation for a commit.

> All commits must to be signed. Your signature certifies that you wrote the patch

Reviewer: ```suggestion
All commits have to be signed. Your signature certifies that you wrote the patch
```

    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

Then you add a line to every git commit message:

    Signed-off-by: Joe Smith <joe.smith@example.com>

Use your real name (sorry, no pseudonyms or anonymous contributions).

> If you set your `user.name` and `user.email` git configs, you can sign your

Reviewer: ```suggestion
If you set your `user.name` and `user.email` in your local git configuration, you can sign your
```

commit automatically with `git commit -s`.

Note: If your git config information is set properly then viewing the `git log`
information for your commit will look something like this:

```
Author: John Smith <john.smith@example.com>
Date:   Thu Feb 2 11:41:15 2018 -0800

    Update README

    Signed-off-by: John Smith <john.smith@example.com>
```

> Notice the `Author` and `Signed-off-by` lines match. If they don't your PR will

Reviewer: ```suggestion
Notice how the `Author` and `Signed-off-by` lines match. If they don't your PR will
```

Question: Does epinio have this automated DC0 check ?

Author: Can we commit with this language and add the DC0 check next week? I'm asking Matt how they set this up.
